src/Particle.js

```diff
@@ -75,7 +83,9 @@ class Particle {
 				username,
 				password,
 				grant_type: 'password',
+				// @ts-ignore
```

Comment:
I'm curious why these ts-ignore are necessary

Reply:
The ones related to `this` are because they are set by doing `Object.assign(this, Defaults, options);` but they have no explicit `this.clientId` initialization so it was complaining of `this` not having said properties.
I thought of adding default values before the object.assign but wasn't sure which values to give it